lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -292,9 +293,13 @@ def headers
       end
 
       def success_from(response)
-        return false if response.include?('error')
+        return false if response.include?('error') || FAILURE_STATUS.include?(response['status'])
 
-        SUCCESS_STATUS.include?(response['status'] || response['transaction']['status'])
+        if response['transaction']
+          SUCCESS_STATUS.include?(response['transaction']['current_status']) || response['transaction']['status_detail'] == 7
+        else
+          response['status']
+        end
```

Comment:
I suggest you can check this other way::

```suggestion
        SUCCESS_STATUS.include?(response['status'] || response.dig('transaction', 'current_status')) ||
        response.dig('transaction', 'status_detail') == 7
```

Reply:
Noted!!! Thanks!!